server/WFCORE-6728-reload-to-stability-level.adoc

```diff
@@ -0,0 +1,155 @@
+
+=== Nice-to-Have Requirements
+// Requirements in this section do not have to be met to merge the proposed functionality.
+// Note: Nice-to-have requirements that don't end up being implemented as part of
+// the work covered by this proposal should be moved to the 'Future Work' section.
+
+
+=== Non-Requirements
+// Use this section to explicitly discuss things that readers might think are required
+// but which are not required.
+
+=== Future Work
+// Use this section to discuss requirements that are not addressed by this proposal
+// but which may be addressed in later proposals.
+This is currently only implemented for standalone servers, so there is no domain mode support. In the future it could be nice to be able to reload the host controllers and possibly the managed servers to different stability levels.
+
+Also, as this is currently mainly for the testsuite to be able to reload, there is no special support in the CLI's reload command for the enhanced reload operation. If this is seen usable for wider use, adding handling to the CLI command to expose the stability parameter could be interesting. The CLI command would need to take into account both the stability level of the server and the user's RBAC role in order to determine whether the enhanced reload operation is visible to the user. If the operation is visible, and the user wants to use the stability paramter, the enhanced reload operation will be called instead of the current reload operation.
+
+== Backwards Compatibility
+
+// Does this enhancement affect backwards compatibility with previously released
+// versions of WildFly?
+// Can the identified incompatibility be avoided?
+No, we are adding a new operation, which is mainly for testing as described above. There are no configuration changes.
+
+=== Default Configuration
+
+No impact
+
+=== Importing Existing Configuration
+
+No impact
+
+=== Deployments
+
+No impact
+
+=== Interoperability
+
+No impact
+
+== Security Considerations
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+RBAC constraints will be configured on the enhanced reload operation to narrow down who can call it.
+
+== Test Plan
+
+Tests will be added to the testsuite/standalone module to ensure that the ServerSetupTasks mentioned in the requirements section reload properly to the desired stability levels.
+
+== Community Documentation
+
+The only user facing functionality here is the enhanced reload operation. The current intent is to keep this a bit hidden, so no community documentation will be added for this.
```

Comment:
s/so no/so, beyond the wildscribe management API documentation, no/g

Reply:
I suggest:

"The only user facing functionality here is the reload-enhanced *operation*. We deliberately don't promote direct use of low level reload operations, as any use of those operations requires the caller to properly handle reconnecting to the reloaded server. Our end-user documentation around reloading should focus on the high level CLI reload *command*, which is out of scope for this proposal. So, beyond the wildscribe management API documentation no community documentation will be added for this."

Reply:
@bstansberry  Thanks, I also added "Additionally, reloading to a different stability level requires great care to make sure the server configuration is compatible with the target stability level." 

@yersan this touches on our discussion in your other comment